packages/@aws-cdk/aws-secretsmanager/lib/secret-target-attachment.ts

```diff
@@ -0,0 +1,71 @@
+import cdk = require('@aws-cdk/cdk');
+import { ISecret, Secret } from './secret';
+import { CfnSecretTargetAttachment } from './secretsmanager.generated';
+
+/**
+ * The type of service or database that's being associated with the secret.
+ */
+export enum AttachmentTargetType {
+  /**
+   * A database instance
+   */
+  Instance = 'AWS::RDS::DBInstance',
+
+  /**
+   * A database cluster
+   */
+  Cluster = 'AWS::RDS::DBCluster'
+}
+
+/**
+ * Options to add a secret attachement to a secret.
+ */
+export interface SecretTargetAttachmentOptions {
+  /**
+   * The id of the target to attach the secret to.
+   */
+  targetId: string;
+
+  /**
+   * The type of the target to attach the secret to.
+   */
+  targetType: AttachmentTargetType;
+}
+
+/**
+ * Construction properties for a SecretAttachement.
+ */
+export interface SecretTargetAttachmentProps extends SecretTargetAttachmentOptions {
+  /**
+   * The secret to attach to the target.
+   */
+  secret: ISecret;
+}
+
+/**
+ * A secret target attachment.
+ */
+export class SecretTargetAttachment extends cdk.Construct {
+  /**
+   * The secret attached to the target.
+   */
+  public readonly secret: ISecret;
+
+  constructor(scope: cdk.Construct, id: string, props: SecretTargetAttachmentProps) {
+    super(scope, id);
+
+    const attachment = new CfnSecretTargetAttachment(this, 'Resource', {
+      secretId: props.secret.secretArn,
+      targetId: props.targetId,
+      targetType: props.targetType
+    });
+
+    // This allows to reference the secret after attachment (dependency). When
+    // creating a secret for a RDS cluster or instance this is the secret that
+    // will be used as the input for the rotation.
+    this.secret = Secret.import(this, 'Secret', {
```

Comment:
It feels like this is something that users might forget, especially since these docs are in implementation notes.

Another idea (not necessarily saying you must do this), how about changing this into:

```ts
class AttachedSecret implements ISecret {
  ...
}
```

So that an `AttachedSecret` can be used anywhere a secret can be used? And if not that, then at least put these notes in a public comment somewhere, seems very relevant to this construct's usage.

Reply:
I like this idea of the `AttachedSecret`